`test_roundtrip_pandas_dataframe_datetime` (tz-aware datetime roundtrip is off by 2h near DST boundaries)

### What is your issue?

What follows is a claude-generated summary of a cause of test failures on `main`. I didn't see an open issue to track this, I'm happy to close if this is redundant.

# CI on `main` is failing: `test_roundtrip_pandas_dataframe_datetime` (tz-aware datetime roundtrip is off by 2h near DST boundaries)

### What is your issue?

`main` has been red on every regular CI run since ~2026-05-12 (last green commit `04d8f655`). On **every** platform/job the failure is the same single Hypothesis property test:

```
properties/test_pandas_roundtrip.py::test_roundtrip_pandas_dataframe_datetime
```

It is not a regression from any one PR — it is a pre-existing latent bug in xarray's timezone-aware datetime handling that Hypothesis happens to surface whenever it draws a far-future, DST-boundary timestamp. Because the seed/example varies, it shows up intermittently, but it has been hitting essentially every run lately and is currently blocking unrelated PRs.

### Falsifying example

```
Falsifying example: test_roundtrip_pandas_dataframe_datetime(
    df=               datetime_col  other_col
    0 2100-03-31 23:00:00-04:00          0,
)

AssertionError: DataFrame.iloc[:, 0] (column name="datetime_col") are different
[left]:  4110235200000000000
[right]: 4110228000000000000   # 7200s = 2 hours earlier
```

### Minimal reproduction

The failing test roundtrips through `Dataset.from_dataframe` → `to_dataframe`. The bug reproduces directly, no Hypothesis needed:

```python
import pandas as pd
import xarray as xr

ts = pd.Timestamp("2100-03-31 23:00:00", tz="America/Halifax")
df = pd.DataFrame({"datetime_col": [ts]}).astype(
    {"datetime_col": pd.DatetimeTZDtype(unit="ns", tz="America/Halifax")}
)

ds = xr.Dataset.from_dataframe(df)
rt = ds.to_dataframe()

print("original :", df["datetime_col"].iloc[0])
print("roundtrip:", rt["datetime_col"].iloc[0])
print("delta    :", df["datetime_col"].iloc[0] - rt["datetime_col"].iloc[0])
```

```
original : 2100-03-31 22:00:00-03:00
roundtrip: 2100-03-31 20:00:00-03:00
delta    : 0 days 02:00:00
```

The roundtripped value is shifted by 2 hours. The shift is timezone-dependent — e.g. `America/New_York` and `America/Halifax` are affected, while `America/Asuncion` and `America/Santiago` roundtrip correctly for the same nominal timestamp — which points at incorrect DST-offset handling for far-future dates rather than a uniform offset error.

### Root cause

This is the same family of bug as the existing tz-aware datetime issues:

- #3291 — `from_dataframe` / `from_series` does not preserve a tz-aware `DatetimeIndex` (the exact code path here)
- #9307 — `.isel()` drops timezone information from a tz-aware index
- #3320 — tz-aware time index cannot be serialized to netCDF
- #11248 — umbrella: "Add tests for timezone aware ExtensionArrays" (notes tz support is new and under-tested)

The property test is doing its job — it caught a real roundtrip defect now that tz-aware ExtensionArray support exists.

### Current state / workaround

There is **no PR dedicated to this**. The only fix in flight is an *incidental* one-line change to the test strategy, bundled into two unrelated open PRs (#11347 scipy pickling, #11348 MultiIndex selection), both narrowing the Hypothesis draw range to avoid far-future DST dates:

```diff
 datetimes = st.datetimes(
-    min_value=pd.Timestamp("1677-09-21T00:12:43.145224193"),
-    max_value=pd.Timestamp("2262-04-11T23:47:16.854775807"),
+    min_value=pd.Timestamp("1970-01-01T00:00:00"),
+    max_value=pd.Timestamp("2037-12-31T23:59:59.999999"),
     timezones=st.just(tz),
 )
```

That only hides the symptom (it stops Hypothesis from drawing the dates that trip the bug); the underlying 2-hour roundtrip error is still present. We should:

1. Land the test-strategy change (or an `xfail`) as a small standalone PR to unblock CI, and
2. Track/fix the real `from_dataframe`/`to_dataframe` tz-roundtrip bug under the tz-aware ExtensionArray work (#11248).

### Environment

```
xarray:     main (repro on 2024.11.1.dev713+g666584993)
pandas:     3.0.3
numpy:      2.4.6
hypothesis: 6.153.2
```

Affected jobs: all OS (Linux/macOS/Windows), py311 + py313, including `no-dask` and `no-numba` — i.e. unrelated to optional deps, consistent with a core datetime bug.

--- a/properties/test_pandas_roundtrip.py
+++ b/properties/test_pandas_roundtrip.py
@@ -125,6 +125,7 @@ def test_roundtrip_pandas_dataframe(df) -> None:
     xr.testing.assert_identical(arr, roundtripped.to_xarray())
 
 
+@pytest.mark.xfail(reason="https://github.com/pandas-dev/pandas/issues/65712")
 @given(df=dataframe_strategy())
 def test_roundtrip_pandas_dataframe_datetime(df) -> None:
     # Need to name the indexes, otherwise Xarray names them 'dim_0', 'dim_1'.

